### PR TITLE
[SLP] Move some of X86 tests to common directory

### PR DESCRIPTION
**Pull request by ElvinaYakubova**
Some of the tests from X86 directory can be generalized for AArch64 to improve its coverage.

### CODE REVIEW

**Review by alexey-bataev — COMMENTED** [review]
These test must be either copied to AArch64 directory, or special directives REQUIRES: x86, aarch64 should be used to avoid false failed tests in some llvm configs

**Comment by sushgokh** [comment]
@alexey-bataev 
>REQUIRES: x86, aarch64
Does not this mean that the supported host architecture must be x86/aarch64 ? We just want to run the tests for x86/aarch64 which is indicated by the -mtriple option and hence, it wont run for other triples.

**Comment by ElvinaYakubova** [comment]
> If the llvm compiled without x86 or aarch64, it still tries to run all the tests. These tests will fail, because llvm won't be able execute them successfully. To prevent their unconditional execution, need to add special directives that checks if the compiler supports given target. Or put them in the corresponding directories, where config files do the same - check if the llvm support given target before running the tests in the given directories.

Thanks for the explanation. I updated the commit with REQUIRES since we don't want to duplicate tests

**Review by alexey-bataev — APPROVED** [review]
LG

File: llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll [hunk 1]
```diff
@@ -1,5 +1,7 @@
 ; NOTE: Assertions have been autogenerated by utils/update_test_checks.py
+; REQUIRES: target={{aarch64|x86_64}}-{{.*}}
```
Comment (fhahn) [inline llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll:2]: Did you check this works as expected?

With this patch merged, those tests are now considered 'unsupported' even when all targets are enabled.

You can check this by applying the patch below and observe that there are no test failures

```
diff --git a/llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll b/llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll
index a709af4cf858..1511dfbe37b6 100644
--- a/llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll
+++ b/llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll
@@ -6,7 +6,6 @@
 define void @test() {
 ; CHECK-LABEL: @test(
 ; CHECK-NEXT:  entry:
-; CHECK-NEXT:    [[CALL37:%.*]] = load i16, ptr poison, align 2
 ; CHECK-NEXT:    br label [[BB:%.*]]
 ; CHECK:       bb:
 ; CHECK-NEXT:    [[CALL:%.*]] = load i16, ptr poison, align 2
```

Comment (ElvinaYakubova) [reply llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll:2]: yes, I checked and locally for me it worked fine
There are some buildbots failures as well, I'll try to figure out what's going on asap

Comment (fhahn) [reply llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll:2]: If it helps, the above is for ARM64 macOS

Comment (ElvinaYakubova) [reply llvm/test/Transforms/SLPVectorizer/alternate-cmp-swapped-pred-parent.ll:2]: @fhahn it should be fixed now https://github.com/llvm/llvm-project/pull/106531

**Pull request merged**